lib/matplotlib/text.py

```diff
@@ -882,8 +882,9 @@ def get_window_extent(self, renderer=None, dpi=None):
             A renderer is needed to compute the bounding box.  If the artist
             has already been drawn, the renderer is cached; thus, it is only
             necessary to pass this argument when calling `get_window_extent`
-            before the first `draw`.  In practice, it is usually easier to
-            trigger a draw first (e.g. by saving the figure).
+            before the first draw.  In practice, it is usually easier to
+            trigger a draw first, e.g. by calling
+            `~.Figure.draw_without_rendering` or ``plt.show()``.
```

Comment:
Saving is usually not a good way to trigger a draw in this case. You'll likely want to modify the figure somehow based on the extent.

Reply:
I'm 50/50 about including `plt.show()` as

```python
plt.ion()
for j in range(5):
    fig, ax = plt.subplots()
    plt.show()
    sleep(5)
```

will not spin the GUI event loop (and hence in some cases not get the actual render to happen) until after the loop is done.

Reply:
OK, but 99.999% of cases, a draw will occur!